tasks/beats-Windows.yml

```diff
@@ -0,0 +1,57 @@
+---
+- set_fact: beat_name="{{ beat }}"
+- name: Windows - Ensure WinPcap is installed before install packetbeat
+  win_chocolatey:
+    name: nmap
+    state: present
+    version: 7.12
+  when: beat_name == "packetbeat"
+
+- name: Windows - Ensure {{ beat }} is installed
+  win_chocolatey:
+    name: "{{ beat }}"
+    state: present
+    version: "{{ beats_version }}"
+  when: use_repository
+  register: beat_install
+  notify: restart beat
+
+- name: Windows - nssm is installed
+  win_chocolatey:
+    name: nssm
+    state: present
+  register: nssm_install
+  when: not use_repository
+
+- set_fact: os_arch="x86"
+
+- set_fact: os_arch="x86_64"
+  when: ansible_architecture == "64-bit"
+
+- name: Windows - Create {{ beat }} dir
+  win_file:
+      path: "{{ beats_home_dir }}"
+      state: directory
+  when: not use_repository
+
+- name: Windows - Download {{ beat }} from url
+  win_get_url: 
+    url: "{% if custom_package_url is defined %}{{ custom_package_url }}{% else %}{{ beats_package_url }}/{{ beat }}/{{ beat }}-{{ beats_version }}-windows-{{ os_arch }}.zip{% endif %}"
```

Comment:
if **use_repository=false**
*url: "{% if custom_package_url is defined %}{{ custom_package_url }}{% else %}{{ beats_package_url }}/{{ beat }}/{{ beat }}-{{ beats_version }}-windows-{{ os_arch }}.zip{% endif %}"* will cause error, an alternate way is 
url: "{% if custom_package_url is defined %}{{ custom_package_url }}**/{{ beat }}/{{ beat }}-{{ beats_version }}-windows-{{ os_arch }}.zip**{% else %}{{ beats_package_url }}/{{ beat }}/{{ beat }}-{{ beats_version }}-windows-{{ os_arch }}.zip{% endif %}"